pkg/measurements/netpol_latency.go

```diff
@@ -0,0 +1,613 @@
+		log.Fatalf("Failed to read response body: %v", err)
+	}
+
+	if err := json.Unmarshal(body, &clusterResults); err != nil {
+		log.Fatalf("Failed to unmarshal results: %v", err)
+	}
+	log.Tracef("clusterResults %v", clusterResults)
+	for pod, results := range clusterResults {
+		for _, res := range results {
+			ar := fmt.Sprintf("%s:%s", pod, res.Address)
+			for _, netpolName := range addrReuse[ar] {
+				connLatency := float64(res.Timestamp.Sub(npCreationTime[netpolName]).Milliseconds())
+				if _, ok := npResults[netpolName]; !ok {
+					npResults[netpolName] = []float64{}
+				}
+				npResults[netpolName] = append(npResults[netpolName], connLatency)
+			}
+		}
+	}
+	resp.Body.Close()
+	for name, npl := range npResults {
+		val, _ := stats.Percentile(npl, 50)
+		p50 := int(val)
+		val, _ = stats.Percentile(npl, 95)
+		p95 := int(val)
+		val, _ = stats.Percentile(npl, 99)
+		p99 := int(val)
+		val, _ = stats.Max(npl)
+		maxVal := int(val)
+		val, _ = stats.Mean(npl)
+		avgVal := int(val)
+		val, _ = stats.Min(npl)
+		minVal := int(val)
+		log.Tracef("%s: 50th: %d 95th: %d 99th: %d min: %d max: %d avg: %d\n", name, p50, p95, p99, minVal, maxVal, avgVal)
+
+		// Use minVal for reporting as ping test tool might have delayed initiating ping tests from some remote addresses.
+		// This can happen when remote pod was busy pinging other pods before trying our network policy pod
+		n.metrics.Store(name, netpolMetric{
+			Name:            name,
+			Timestamp:       npCreationTime[name],
+			MetricName:      netpolLatencyMeasurement,
+			MinReadyLatency: time.Duration(minVal),
+			ReadyLatency:    time.Duration(maxVal),
+			UUID:            globalCfg.UUID,
+			Metadata:        factory.metadata,
+			JobName:         factory.jobConfig.Name,
+		})
+	}
+}
+
+// Read network policy object template
+func readTemplate(o kconfig.Object) ([]byte, error) {
+	var err error
+	var f io.Reader
+
+	e := embed.FS{}
+	if embedFS == e {
+		f, err = kutil.ReadConfig(o.ObjectTemplate)
+	} else {
+		objectTemplate := path.Join(embedFSDir, o.ObjectTemplate)
+		f, err = kutil.ReadEmbedConfig(embedFS, objectTemplate)
+	}
+	if err != nil {
+		log.Fatalf("Error reading template %s: %s", o.ObjectTemplate, err)
+	}
+	t, err := io.ReadAll(f)
+	if err != nil {
+		log.Fatalf("Error reading template %s: %s", o.ObjectTemplate, err)
+	}
+	return t, err
+}
+
+func getObjectType(o kconfig.Object, t []byte) string {
+	// Deserialize YAML
+	cleanTemplate, err := prepareTemplate(t)
+	if err != nil {
+		log.Fatalf("Error preparing template %s: %s", o.ObjectTemplate, err)
+	}
+	uns := &unstructured.Unstructured{}
+	yamlToUnstructured(o.ObjectTemplate, cleanTemplate, uns)
+	return uns.GetKind()
+}
+
+// Start network policy measurement
+func (n *netpolLatency) start(measurementWg *sync.WaitGroup) error {
+	defer measurementWg.Done()
+	// Use proxyRoute to communicate with proxy pod
+	proxyRoute = n.config.Config["proxyRoute"]
+	// Skip latency measurement for 1st job which creates only pods
```

Comment:
Let's not assume that there's going to be a first job

```suggestion
	// Skip latency measurement in namespaces with this label
```